kernel/syscall.c

```diff
@@ -6605,3 +6610,30 @@ long myst_syscall_unload_symbols(void)
     long params[6] = {0};
     return myst_tcall(MYST_TCALL_UNLOAD_SYMBOLS, params);
 }
+
+long myst_syscall_pause(void)
+{
+    long ret = 0;
+    myst_thread_t* thread = myst_thread_self();
+    // Set futex unavilable. This is to make sure the futex which was made
+    // available by signals delivered before calling pause won't affect the
```

Comment:
Does a futex wake set the futex value to 1?

Reply:
Oh it looks like `__sync_val_compare_and_swap(pause_futex, 0, 1)` in `myst_signal_deliver` does that.

Reply:
Yes @vtikoo that is correct.